src/macaron/__main__.py

```diff
@@ -62,25 +72,45 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
 
     run_config = {}
 
-    if analyzer_single_args.repo_path:
-        # Generate a config dict similar to a config read from user yaml file.
+    if analyzer_single_args.config_path:
+        # Get user config from yaml file
+        run_config = YamlLoader.load(analyzer_single_args.config_path)
+        if run_config is None:
+            sys.exit(os.EX_DATAERR)
+    else:
+        repo_path = analyzer_single_args.repo_path
+        purl = analyzer_single_args.package_url
+        branch = analyzer_single_args.branch
+        digest = analyzer_single_args.digest
+
+        if repo_path and purl and not (branch and digest):
+            # To provide the purl together with the repository path, the user must specify the branch and commit
+            # digest.
+            logger.error(
+                "Please provide branch and commit digest for the repo at %s because a PURL %s is also provided",
```

Comment:
```suggestion
               "Please provide the branch and commit digest for the repo at %s that matches to the PURL string %s.",
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/401/commits/f66b0be2bf3c2556eed5885315d39cd66b792306